Repairing programs

bn0785ac opened the pull request:
Troubles:
* Cinelerra is single-threaded (*with an i3/i5* **may be acceptable**,  with a fx 8350/ryzen *1400,2400G,1500X/ ryzen 3*   its bad, but with a 1600/2600/1700/2700X/1800, i9, i7, i5 or threadripper* **is really annoying**)


* Libreoffice doesn't open *global*

* Lobase can't close *only lobase*

bn0785ac commented:
This pull fixes #1893 #1879

Fred-Barclay commented on etc/natron.profile:
```diff
@@ -12,7 +12,7 @@ noblacklist /opt/natron
 include /etc/firejail/disable-common.inc
 include /etc/firejail/disable-devel.inc
-include /etc/firejail/disable-interpreters.inc
+#include /etc/firejail/disable-interpreters.inc
```
Hi @bn0785ac Do you know which language(s) natron needs access to? We can allow access to those while still blocking access to the others in disable-interpreters.

bn0785ac replied:
python :)

Fred-Barclay replied:
Thanks!

bn0785ac replied:
Fixed (natron runs but i got crazy logs)


TESTING warning: noblacklist /usr/lib/python3* not matched by a proper blacklist command in disable*.inc

Fred-Barclay commented on etc/cin.profile:
```diff
@@ -18,15 +18,15 @@ ipc-namespace
 noroot
 protocol unix
-seccomp
+#seccomp
```
I'm not sure we should disable some of these (especially `seccomp`). @SkewedZeppelin any thoughts?

SkewedZeppelin replied:
Agreed.

bn0785ac replied:
rendertime 

13:55  with     seccom
13:05  without it

Fred-Barclay commented on etc/libreoffice.profile:
```diff
@@ -27,13 +27,13 @@ apparmor
 noroot
 notv
-protocol unix,inet,inet6
+#protocol unix,inet,inet6
```
Why does this line need commenting?

bn0785ac replied:
i got crazy errors at libreoffice

smitsohu replied:
@bn0785ac We still try to find a different solution to the problem with Libreoffice in #1703. `nodbus` (for KDE in general) and `apparmor` (for Kubuntu only) probably needs to be disabled, yes.

Vincent43 replied:
What breaks apparmor?

bn0785ac replied:
REAL errors :dagger:

smitsohu replied:
@Vincent43 D-Bus. Means issues are probably limited to Kubuntu, see #1703.

bn0785ac replied:
no,  it happens here (manjaro *unstable* kde)


manjaro unstable = arch but with a 24 hours delay for update

Fred-Barclay replied:
Can you paste the errors you get and describe what you see?
Thanks!

SkewedZeppelin commented on etc/cin.profile:
```diff
@@ -18,15 +18,15 @@ ipc-namespace
 shell none
 
-private-bin cin,ffmpeg
+#private-bin cin,ffmpeg
```
This is okay

bn0785ac replied:
I'll run a small bench with seccomp enabled & disabled

bn0785ac replied:
rendertime 

13:55  with     seccom
13:05  without it

smitsohu replied:
`seccomp` is indeed not for free performance-wise. The penalty is small, but when you do heavy computing, like rendering videos in your case, it could be measurable.

bn0785ac replied:
yeah :)   

we need say if this 7% gap  its acceptable (i've a fx 8350),  MAY be small (2-3)  at an i5,  <1 at an i3 or memecpus,  huge (14) with a 2700X  and INSANELY BAD (24-36) with a TR/i9

smitsohu replied:
@bn0785ac Since `seccomp` is an important component of the sandbox, I'm not sure if disabling it is the right way. But probably there should be a comment in the profile, so other users can find the solution easily when they run into this issue.

I assume `nogroups` has a similar effect?

bn0785ac replied:
i like this **accepted**  

with nogroups i found a harder gap  
the same file spent 16:05  (same setup, fx 8350-4.1) **idk if with an i3-i5 you'll notice a ligher gap, or if with a  r5-2600/600/ ryzen 7/TR/i9 you'll notice an insanely evil gap**

>15%

bn0785ac replied:
**edit**: i've added this.

* re-enable seccomp

* add a comment: its an estimated due to  my benchs,  **as i dont have other pcs** i can't confirm that an i3 really has a 2% gap, a 2700X  has a 14-28% gap,  or a 1950X/7980XE really has **OFFENSIVE gaps** (+50%)

bn0785ac commented:
@smitsohu I've made all that you've asked 

* reduced gap + fixing multithread issues, and i'm still keeping seccomp enabled

* i've added a small warning about   the seccomp performance gap 

* fixed natron (with a better fix)

smitsohu commented:
@bn0785ac Thanks!

bn0785ac commented:
@smitsohu @SkewedZeppelin @Fred-Barclay 
can somebody accept it?

* cin keeps seccomp  (but i've recovered multithreading **its mandatory for video editing**)

* libreoffice its fixed (linux-kde)

* natron works (multithread )

Fred-Barclay commented:
@bn0785ac Have you been able to make any progress getting that info for LibreOffice? :smiley:

Fred-Barclay commented:
@smitsohu As far as I know we may not need `#nodbus` in this PR. Other than that we should be good to go!

Fred-Barclay commented:
Alright, here we go!
